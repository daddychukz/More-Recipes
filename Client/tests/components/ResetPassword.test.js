import React from 'react';
import { mount, shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import sinon from 'sinon';
import render from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ConnectedResetPassword,
{ ResetPassword } from '../../src/components/pages/ResetPassword';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const initialState = {
  user: {}
};
const store = mockStore(initialState);

const props = {
  validateToken: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  match: {
    params: {
      token: ''
    }
  }
};

const mountedWrapper = mount(
  <BrowserRouter>
    <Provider store={store}>
      <ConnectedResetPassword {...props} />
    </Provider>
  </BrowserRouter>
);

describe('MyFavorite test suite', () => {
  it('renders MyFavorite component without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('ResetPassword').length).toBe(1);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <BrowserRouter >
        <Provider store={store}>
          <ResetPassword {...props} />
        </Provider>
      </BrowserRouter>);
    expect(tree).toMatchSnapshot();
  });
});
