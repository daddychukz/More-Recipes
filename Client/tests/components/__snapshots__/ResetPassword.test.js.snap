// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFavorite test suite should match component snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md bg-dark navbar-dark fixed-top"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        <h3
          id="logo"
        >
          More-Recipes
        </h3>
      </a>
      <button
        className="navbar-toggler"
        data-target="#navbarCollapse"
        data-toggle="collapse"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarCollapse"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <header
    style={
      Object {
        "background": "#333333",
        "paddingTop": "67px",
      }
    }
  />
  <section
    className="d-flex flex-row align-items-center justify-content-center obj-color"
    style={
      Object {
        "height": "48vw",
        "width": "100vw",
      }
    }
  >
    <br />
    <div>
      <div
        className="loader"
      />
    </div>
  </section>
</div>
`;
